plugin.js

```diff
@@ -0,0 +1,6 @@
+var ghPages = require('./lib/index')
+var path = require('path')
+
+module.exports = function (pluginConfig, config, callback) {
+  ghPages.publish(path.join(__dirname, config.basePath), config, callback)
```

Comment:
I think it would be unexpected to have `basePath` be resolved relative to the `plugin.js` directory.  I imagine it would make more sense to join it to `process.cwd()`.  And you should be able to just pass it along directly to get that behavior.